lib/capybara/queries/text_query.rb

```diff
@@ -42,6 +53,15 @@ def negative_failure_message
       def valid_keys
         COUNT_KEYS + [:wait]
       end
+
+      def visible?
+        @type == :visible or (@type.nil? and Capybara.ignore_hidden_elements)
```

Comment:
this would need to take Capybara.visible_text_only into account to match the behavior of Node#text